Fix bug in main screen tabs state management

mauriciocolli opened the pull request:
Tabs were not being destroyed/restored correctly due to a call to a method that populated the view pager before it even had a chance of restoring itself.

The solution was to null out the adapter before calling that method so the view pager will postpone the populating process.

Stypox commented on app/src/main/java/org/schabi/newpipe/fragments/MainFragment.java:
```diff
@@ -136,16 +142,23 @@ public boolean onOptionsItemSelected(MenuItem item) {
             pagerAdapter = new SelectedTabsPagerAdapter(requireContext(), getChildFragmentManager(), tabsList);
         }
         // Clear previous tabs/fragments and set new adapter
         viewPager.setAdapter(pagerAdapter);
```
Couldn't this be put inside the `if (!wasAdapterNull)`, too?

mauriciocolli commented on app/src/main/java/org/schabi/newpipe/fragments/MainFragment.java:
```diff
@@ -136,16 +142,23 @@ public boolean onOptionsItemSelected(MenuItem item) {
         tabsList.addAll(tabsManager.getTabs());
 
+        boolean wasAdapterNull = pagerAdapter == null;
         if (pagerAdapter == null || !pagerAdapter.sameTabs(tabsList)) {
```
I think the name was confusing at first and I decided to keep things clear, however, later on I renamed to something more readable and indeed, it would be better to do this.

mauriciocolli commented:
Changed the way I fixed the bug.

There's also the fact that [this class is deprecated](https://developer.android.com/reference/androidx/fragment/app/FragmentStatePagerAdapter) and the [adapter from the ViewPager2](https://developer.android.com/reference/androidx/viewpager2/adapter/FragmentStateAdapter) shoud be used instead, will do the migration in the future.
